data-pipeline/src/concentrator/aggregation.rs

```diff
@@ -0,0 +1,497 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+//! This module implement the logic for stats aggregation into time buckets and stats group.
+//! This includes the aggregation key to group spans together and the computation of stats from a
+//! span.
+use datadog_trace_protobuf::pb;
+use datadog_trace_utils::trace_utils::has_top_level;
+use ddcommon::tag::Tag;
+use std::collections::HashMap;
+
+const TAG_STATUS_CODE: &str = "http.status_code";
+const TAG_SYNTHETICS: &str = "synthetics";
+const TAG_SPANKIND: &str = "span.kind";
+const TAG_ORIGIN: &str = "_dd.origin";
+
+/// This struct represent the key used to group spans together to compute stats.
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Default)]
+pub struct AggregationKey {
+    pub resource_name: String,
+    pub service_name: String,
+    pub operation_name: String,
+    pub span_type: String,
+    pub span_kind: String,
+    pub http_status_code: u32,
+    pub is_synthetics_request: bool,
+    pub peer_tags: Vec<Tag>,
+    pub is_trace_root: bool,
+}
```

Comment:
This is not for this PR, but maybe the `AggregationKey` should cache a precomputed hash value (since it's immutable anyway), and implement `Hash` by itself.